spinner/spinner.go

```diff
@@ -153,10 +153,11 @@ func (s *Spinner) Run() error {
 		return s.ctx.Err()
 	}
 
+	var actionErr error
 	p := tea.NewProgram(s, tea.WithContext(s.ctx), tea.WithOutput(os.Stderr))
 	if s.ctx == nil {
 		go func() {
-			s.action()
+			actionErr = s.action()
 			p.Quit()
```

Comment:
I don't understand how the code there could send an error to a code that is out of the go func, so without having a channel and wait for it.

There is something strange to me in the existing code. And I'm afraid the changes are making things worse

Reply:
BTW, I hope `p.Quit` doesn't call `os.Exit`otherwise the code out of the go routine (I'm already doubtful about) will never be executed

Reply:
i rewrote this in the latest commit